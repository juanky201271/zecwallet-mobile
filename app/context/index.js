import { defaultAppContextLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppContextLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppContextLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppContextLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
