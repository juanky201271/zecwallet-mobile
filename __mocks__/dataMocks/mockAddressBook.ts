import { AddressBookFileClass } from '../../app/AppState';

export const mockAddressBook: AddressBookFileClass[] = [
  {
    label: 'pepe',
    address: 'u1234567890_____________',
  },
  {
    label: 'lolo',
    address: 'u0987654321_____________',
  },
];
