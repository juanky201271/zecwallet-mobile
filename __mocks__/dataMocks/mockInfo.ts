import { ChainNameEnum, CurrencyNameEnum, InfoType } from '../../app/AppState';
import { serverUris } from '../../app/uris';

export const mockInfo: InfoType = {
  chainName: ChainNameEnum.mainChainName,
  serverUri: serverUris(() => {})[0].uri,
  latestBlock: 2000000,
  connections: 0,
  version: 'server_version',
  verificationProgress: 0,
  currencyName: CurrencyNameEnum.ZEC,
  solps: 0,
  zingolib: 'zingolib_version',
};
