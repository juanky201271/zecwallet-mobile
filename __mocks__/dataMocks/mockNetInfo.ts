import { NetInfoStateType } from '@react-native-community/netinfo';
import { NetInfoType } from '../../app/AppState';

export const mockNetInfo: NetInfoType = {
  isConnected: true,
  type: NetInfoStateType.wifi,
  isConnectionExpensive: false,
};
