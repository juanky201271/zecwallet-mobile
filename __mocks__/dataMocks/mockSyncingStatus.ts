import { SyncingStatusClass } from '../../app/AppState';

export const mockSyncingStatus: SyncingStatusClass = new SyncingStatusClass();

mockSyncingStatus.syncID = 1;
mockSyncingStatus.inProgress = true;
mockSyncingStatus.currentBatch = 5;
mockSyncingStatus.totalBatches = 50;
mockSyncingStatus.currentBlock = 1800100;
mockSyncingStatus.lastBlockWallet = 1800000;
mockSyncingStatus.lastBlockServer = 1900100;
mockSyncingStatus.secondsPerBatch = 122;
mockSyncingStatus.processEndBlock = 1600100;

export default mockSyncingStatus;
