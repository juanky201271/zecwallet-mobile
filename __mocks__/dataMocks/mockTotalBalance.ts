import { TotalBalanceClass } from '../../app/AppState';

export const mockTotalBalance: TotalBalanceClass = {
  transparentBal: 0.12345678,
  privateBal: 0.4,
  spendablePrivate: 0.2,
  orchardBal: 0.6,
  spendableOrchard: 0.3,
  total: 1.12345678,
};
