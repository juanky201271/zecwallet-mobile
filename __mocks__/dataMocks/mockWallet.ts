import { WalletType } from '../../app/AppState';

export const mockWallet: WalletType = {
  seed: 'pepe lolo titi',
  ufvk: 'uview1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890',
  birthday: 1900100,
};
