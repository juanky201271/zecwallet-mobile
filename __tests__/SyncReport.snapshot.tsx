/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import SyncReport from '../components/SyncReport';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { mockWallet } from '../__mocks__/dataMocks/mockWallet';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockInfo } from '../__mocks__/dataMocks/mockInfo';
import { mockTotalBalance } from '../__mocks__/dataMocks/mockTotalBalance';
import mockSyncingStatus from '../__mocks__/dataMocks/mockSyncingStatus';
import { mockNetInfo } from '../__mocks__/dataMocks/mockNetInfo';

const NetInfoStateType = {
  unknown: 'unknown',
  none: 'none',
  cellular: 'cellular',
  wifi: 'wifi',
  bluetooth: 'bluetooth',
  ethernet: 'ethernet',
  wimax: 'wimax',
  vpn: 'vpn',
  other: 'other',
};

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => ({
  RNCNetInfo: () => {
    const RN = jest.requireActual('react-native');

    RN.NativeModules.RNCNetInfo = {
      execute: jest.fn(() => '{}'),
    };

    return RN;
  },
  NetInfoStateType: NetInfoStateType,
}));
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component SyncReport - test', () => {
  //snapshot test
  const state = defaultAppContextLoaded;
  state.translate = mockTranslate;
  state.info = mockInfo;
  state.totalBalance = mockTotalBalance;
  state.wallet = mockWallet;
  state.syncingStatus = mockSyncingStatus;
  state.netInfo = mockNetInfo;
  const onClose = jest.fn();
  test('SyncReport - snapshot', () => {
    const sync = render(
      <ContextAppLoadedProvider value={state}>
        <SyncReport closeModal={onClose} />
      </ContextAppLoadedProvider>,
    );
    expect(sync.toJSON()).toMatchSnapshot();
  });
});
