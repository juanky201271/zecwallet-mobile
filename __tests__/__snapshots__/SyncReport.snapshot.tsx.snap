// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SyncReport - test SyncReport - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "display": "flex",
          "minHeight": 25,
          "paddingBottom": 0,
          "paddingTop": 10,
          "zIndex": -1,
        }
      }
      testID="header"
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 12,
          }
        }
      />
    </View>
    <View
      style={
        {
          "left": 0,
          "padding": 11.5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
          }
        }
      />
    </View>
    <View
      style={
        {
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 38,
            "resizeMode": "contain",
            "width": 38,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="rgb(0, 122, 255)"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  9001,
                ],
                "f053",
                "M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          size={30}
          style={
            {
              "marginHorizontal": 10,
            }
          }
        />
      </View>
      <Text
        selectable={true}
        style={
          {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        text translated
      </Text>
      <View
        style={
          {
            "height": 30,
            "marginHorizontal": 10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      {
        "maxHeight": "85%",
      }
    }
    testID="syncreport.scroll-view"
  >
    <View>
      <View
        style={
          {
            "display": "flex",
            "marginBottom": 30,
            "marginHorizontal": 20,
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Sync ID
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              1 - (text translated)
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "height": 2,
              "marginBottom": 10,
              "marginTop": 15,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1600099
          </Text>
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1900100
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "height": 10,
              }
            }
          />
          <View
            style={
              {
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
              }
            }
          />
        </View>
        <View
          style={
            {
              "borderBottomColor": "rgb(0, 122, 255)",
              "borderBottomWidth": 2,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "lightyellow",
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "100%",
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "orange",
                "borderRightColor": "orange",
                "borderRightWidth": 1,
                "height": 10,
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            text translated
          </Text>
          <View
            style={
              {
                "backgroundColor": "lightyellow",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            1600099text translated100.00%
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            text translated
          </Text>
          <View
            style={
              {
                "backgroundColor": "orange",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
            testID="syncreport.syncednow"
          >
            200001text translated100.00%
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "height": 2,
              "marginTop": 15,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
              testID="syncreport.currentbatch"
            >
              text translated5text translated50
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
              testID="syncreport.blocksperbatch"
            >
              100
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "marginTop": 20,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "opacity": 0.65,
                }
              }
            >
              text translated
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                selectable={true}
                style={
                  {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "opacity": 1,
                  }
                }
              >
                122
              </Text>
            </View>
          </View>
          <ActivityIndicator
            color="rgb(0, 122, 255)"
            size="large"
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "rgb(0, 122, 255)",
              "height": 2,
              "marginTop": 10,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text translated
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              selectable={true}
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              text translated1800100text translated1900100
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
